xmtp_mls/src/groups/mod.rs

```diff
@@ -706,6 +714,36 @@ impl<ScopedClient: ScopedGroupClient> MlsGroup<ScopedClient> {
         Ok(message_id)
     }
 
+    /// Helper function to extract queryable content fields from a message
+    fn extract_queryable_content_fields(message: &[u8]) -> QueryableContentFields {
+        let default = QueryableContentFields {
+            content_type: ContentType::Unknown,
+            version_major: 0,
+            version_minor: 0,
+            authority_id: "unknown".to_string(),
+        };
+
+        // Return early with default if decoding fails or type is missing
+        let content_type_id = match EncodedContent::decode(message)
+            .map_err(|e| tracing::debug!("Failed to decode message as EncodedContent: {}", e))
+            .ok()
+            .and_then(|content| content.r#type)
+        {
+            Some(type_id) => type_id,
```

Comment:
Any reason we can't `impl TryFrom<EncodedContent> for QueryableContentFields`? Because then you could 

```rust
EncodedContent::decode(msg)
    .inspect_err(|e| tracing::debug!(e))
    .and_then(|content| content.r#type)
    .map(QueryableContentFields::try_from)
    .unwrap_or_default()
```

which is more readable

Reply:
thanks this was really helpful. updated here with some more tips from clippy to make it even more concise 💯 
https://github.com/xmtp/libxmtp/pull/1435/commits/fbed7ce02c68dbcb639e5d3cee55e35871299fd1